### PR TITLE
Service create form single container base

### PR DESCRIPTION
**Pull request by mlunoe**
---
~~⚠️ Needs this PR to function well: https://github.com/dcos/dcos-ui/pull/1345~~ Marged

---

This PR introduces the single container view and reducers.

~~The GPU field does nothing as of now. Had trouble finding a way for it to remove the field if it was empty, which had quite an impact on the experience of trying to set a container, so decided to leave it out from the reducers.~~ GPU Field removed for now. Will be introduced later

### CODE REVIEW

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    let {data, errors} = this.props;
+    // console.log(JSON.stringify(data));
+    let advancedSettingsIcon = 'triangle-right';
+    if (this.state.showAdvancedSettings) {
+      advancedSettingsIcon = 'triangle-down';
+    }
+
+    let {container = {}} = data;
+    let type = container.type || MESOS;
+    let typeKey = type && type.toLowerCase();
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Done! I've made the change to fix the container type key as `docker`.

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+          scrollContainer=".gm-scroll-view">
+            <Icon color="grey" id="ring-question" size="mini" family="mini" />
+        </Tooltip>
+      </FieldLabel>
+    );
+  }
+
+  render() {
+    let {data, errors} = this.props;
+    // console.log(JSON.stringify(data));
```
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: This is not needed let's remove this.

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        <p>
+          <a onClick={this.toggleAdvancedSettings}>
+            <Icon
+              id={advancedSettingsIcon}
+              color="purple"
+              family="mini"
+              size="mini" />
+            Advanced Service Settings
+          </a>
+        </p>
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: Let's not use `p` tags for the spacing. We'll probably want to create a new component for this:

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        </div>
+
+        <p>
+          <a onClick={this.toggleAdvancedSettings}>
+            <Icon
+              id={advancedSettingsIcon}
+              color="purple"
+              family="mini"
+              size="mini" />
+            Advanced Service Settings
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: This text should not be underlined and should have the cursor pointer. Also there should be margin on the icon. As I said in another comment, we should probably create a new component for this to ensure we use it consistently in the future.

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        <div className="form-row-element">
+          <h2 className="form-header flush-top short-bottom">
+            Services
+          </h2>
+          <p>
+            Configure your service below. Start by giving your service a name.
+          </p>
+        </div>
+
+        <div className="flex row">
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: I thought we created a component for this?

**Comment by jfurrow** [comment]
Do you know when we plan to address the styling for labels? I just want to make sure it doesn't get left for the last minute.

**Comment by jfurrow** [comment]
In general the form's spacing doesn't match the mockup. I think we should create a `FormSection` component which will give us the right spacing. I'm fine with this being in a separate PR, but let's create a JIRA so that the work doesn't get forgotten about:

**Comment by mlunoe** [comment]
About this: https://github.com/dcos/dcos-ui/pull/1346#issuecomment-260723075
@ashenden has a task on Canvas here: https://github.com/mesosphere/canvas/issues/45

**Comment by mlunoe** [comment]
Had a conversation offline with @jfurrow to create tasks for the following:
- Label `p`s not being uppercased
- Create a component for `flex row` divs and add column option to `FormGroup` (will assign @Poltergeist to this task)
- Create a FieldSet (<fieldset />) component to control bottom margin in the form
- Create a CollapsingSection component for the advanced section bits

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js (hunk 5 above)
Comment (mlunoe) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: This will be handled in a separate PR

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js (hunk 4 above)
Comment (mlunoe) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: This will be handled in a separate PR

**Comment by jfurrow** [comment]
Nice work! :shipit:

**Comment by philipnrmn** [comment]
Are we holding merge on this for any reason?

**Comment by philipnrmn** [comment]
Oh.
